src/main/java/io/supertokens/multitenancy/Multitenancy.java

```diff
@@ -44,32 +46,14 @@
 
 public class Multitenancy extends ResourceDistributor.SingletonResource {
 
-    @TestOnly
-    public static boolean addNewOrUpdateAppOrTenant(Main main, TenantIdentifier sourceTenant, TenantConfig newTenant)
-            throws DeletionInProgressException, CannotModifyBaseConfigException, BadPermissionException,
-            StorageQueryException, FeatureNotEnabledException, IOException, InvalidConfigException,
-            InvalidProviderConfigException, TenantOrAppNotFoundException {
-        return addNewOrUpdateAppOrTenant(main, sourceTenant, newTenant, false);
-    }
-
-    public static boolean addNewOrUpdateAppOrTenant(Main main, TenantIdentifier sourceTenant, TenantConfig newTenant,
-                                                    boolean shouldPreventDbConfigUpdate)
-            throws DeletionInProgressException, CannotModifyBaseConfigException, BadPermissionException,
-            StorageQueryException, FeatureNotEnabledException, IOException, InvalidConfigException,
-            InvalidProviderConfigException, TenantOrAppNotFoundException {
+    public static void checkPermissionsForCreateUpdateOrDelete(Main main, TenantIdentifier sourceTenant,
```

Comment:
function needs to be renamed to remove the "Delete" postfix

Reply:
Done